### PR TITLE
fix #804 on 4.2 branch

### LINKED ISSUE #804: Some tests fail on Windows due to different DIRECTORY_SEPERATOR
```
C:\xampp\php\php.exe -dxdebug.coverage_enable=1 "C:/Users/Alexander Jank/Projekte/ApiGen/vendor/phpunit/phpunit/phpunit" --coverage-clover "C:\Users\Alexander Jank\.PhpStorm2016.3\system\coverage\ApiGen$Unnamed.coverage" --configuration "C:\Users\Alexander Jank\Projekte\ApiGen\phpunit.xml.dist" --teamcity
Testing started at 22:35 ...
PHPUnit 6.0.8 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.1.1 with Xdebug 2.5.1
Configuration: C:\Users\Alexander Jank\Projekte\ApiGen\phpunit.xml.dist


Method Mockery_5_Symfony_Component_Console_Output_OutputInterface::writeln() does not exist on this mock object
 C:\Users\Alexander Jank\Projekte\ApiGen\src\Console\Command\GenerateCommand.php:219
 C:\Users\Alexander Jank\Projekte\ApiGen\tests\MethodInvoker.php:21
 C:\Users\Alexander Jank\Projekte\ApiGen\tests\Console\Command\GenerateCommandExecuteTest.php:54
 

Failed asserting that two strings are identical.
Expected :C:\Users\Alexander Jank\Projekte\ApiGen\tests/temp/38884/api
Actual   :C:/Users/Alexander Jank/Projekte/ApiGen/tests/temp/38884/Api
 <Click to see difference>

 C:\Users\Alexander Jank\Projekte\ApiGen\tests\Console\Command\GenerateCommandPrepareOptionsTest.php:55
 

Failed asserting that two strings are identical.
Expected :C:\Users\Alexander Jank\Projekte\ApiGen\src
Actual   :C:/Users/Alexander Jank/Projekte/ApiGen/src
 <Click to see difference>

 C:\Users\Alexander Jank\Projekte\ApiGen\tests\Console\Command\GenerateCommandPrepareOptionsTest.php:70
 

Failed asserting that two strings are identical.
Expected :entities/Category.php
Actual   :ntities/Category.php
 <Click to see difference>

 C:\Users\Alexander Jank\Projekte\ApiGen\tests\Generator\Resolvers\RelativePathResolverTest.php:66
 

Could not determine "C:\Users\Alexander Jank\Projekte\ApiGen\tests\Generator\TemplateGenerators\SourceCodeSource\SomeClass.php" relative path
 C:\Users\Alexander Jank\Projekte\ApiGen\src\Generator\Resolvers\RelativePathResolver.php:43
 C:\Users\Alexander Jank\Projekte\ApiGen\src\Generator\TemplateGenerators\SourceCodeGenerator.php:127
 C:\Users\Alexander Jank\Projekte\ApiGen\src\Generator\TemplateGenerators\SourceCodeGenerator.php:84
 C:\Users\Alexander Jank\Projekte\ApiGen\tests\Generator\TemplateGenerators\SourceCodeGeneratorTest.php:72
 

Could not determine "C:\Users\Alexander Jank\Projekte\ApiGen\tests\Generator\TemplateGenerators\SourceCodeSource\SomeClass.php" relative path
 C:\Users\Alexander Jank\Projekte\ApiGen\src\Generator\Resolvers\RelativePathResolver.php:43
 C:\Users\Alexander Jank\Projekte\ApiGen\src\Generator\TemplateGenerators\SourceCodeGenerator.php:127
 C:\Users\Alexander Jank\Projekte\ApiGen\tests\MethodInvoker.php:21
 C:\Users\Alexander Jank\Projekte\ApiGen\tests\Generator\TemplateGenerators\SourceCodeGeneratorTest.php:85
 

This test did not perform any assertions

This test did not perform any assertions

This test did not perform any assertions

Failed asserting that two strings are identical.
Expected :C:\Users\Alexander Jank\Projekte\ApiGen\tests/temp/38884/api/index.html
Actual   :C:/Users/Alexander Jank/Projekte/ApiGen/tests/temp/38884/Api/index.html
 <Click to see difference>

 C:\Users\Alexander Jank\Projekte\ApiGen\tests\Templating\TemplateNavigatorTest.php:73
 

Failed asserting that two strings are identical.
Expected :C:\Users\Alexander Jank\Projekte\ApiGen\tests/temp/38884/api/namespace-MyNamespace.html
Actual   :C:/Users/Alexander Jank/Projekte/ApiGen/tests/temp/38884/Api/namespace-MyNamespace.html
 <Click to see difference>

 C:\Users\Alexander Jank\Projekte\ApiGen\tests\Templating\TemplateNavigatorTest.php:91
 

Failed asserting that two strings are identical.
Expected :C:\Users\Alexander Jank\Projekte\ApiGen\tests/temp/38884/api/package-MyPackage.html
Actual   :C:/Users/Alexander Jank/Projekte/ApiGen/tests/temp/38884/Api/package-MyPackage.html
 <Click to see difference>

 C:\Users\Alexander Jank\Projekte\ApiGen\tests\Templating\TemplateNavigatorTest.php:100
 

Failed asserting that two strings are identical.
Expected :C:\Users\Alexander Jank\Projekte\ApiGen\tests/temp/38884/api/class-SomeClass.html
Actual   :C:/Users/Alexander Jank/Projekte/ApiGen/tests/temp/38884/Api/class-SomeClass.html
 <Click to see difference>

 C:\Users\Alexander Jank\Projekte\ApiGen\tests\Templating\TemplateNavigatorTest.php:112
 

Failed asserting that two strings are identical.
Expected :C:\Users\Alexander Jank\Projekte\ApiGen\tests/temp/38884/api/constant-SomeConstant.html
Actual   :C:/Users/Alexander Jank/Projekte/ApiGen/tests/temp/38884/Api/constant-SomeConstant.html
 <Click to see difference>

 C:\Users\Alexander Jank\Projekte\ApiGen\tests\Templating\TemplateNavigatorTest.php:124
 

Failed asserting that two strings are identical.
Expected :C:\Users\Alexander Jank\Projekte\ApiGen\tests/temp/38884/api/function-SomeFunction.html
Actual   :C:/Users/Alexander Jank/Projekte/ApiGen/tests/temp/38884/Api/function-SomeFunction.html
 <Click to see difference>

 C:\Users\Alexander Jank\Projekte\ApiGen\tests\Templating\TemplateNavigatorTest.php:136
 

Failed asserting that two strings are identical.
Expected :C:\Users\Alexander Jank\Projekte\ApiGen\tests/temp/38884/api/source-code-SomeClass.html
Actual   :C:/Users/Alexander Jank/Projekte/ApiGen/tests/temp/38884/Api/source-code-SomeClass.html
 <Click to see difference>

 C:\Users\Alexander Jank\Projekte\ApiGen\tests\Templating\TemplateNavigatorTest.php:148
 

Failed asserting that two strings are identical.
Expected :C:\Users\Alexander Jank\Projekte\ApiGen\tests\Theme/ThemeConfigPathResolverSource/config.neon
Actual   :C:\Users\Alexander Jank\Projekte\ApiGen\tests\Theme/ThemeConfigPathResolverSource//config.neon
 <Click to see difference>

 C:\Users\Alexander Jank\Projekte\ApiGen\tests\Theme\ThemeConfigPathResolverTest.php:17
 
Der Befehl "chmod" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.

Failed asserting that exception of type "\ApiGen\Utils\Neon\Exceptions\FileNotReadableException" is thrown.


Time: 1.21 minutes, Memory: 68.00MB


ERRORS!
Tests: 531, Assertions: 794, Errors: 3, Failures: 12, Risky: 3.

Generating code coverage report in Clover XML format ... done

Process finished with exit code 2
```

### PR DESCRIPTION
**Pull request by luka-zitnik**
Hi,

Just to get myself familiar with the sourcecode.

I have three concerns;
1. Should have this be done for the upcoming v5?
2. Is `FileSystem::normalizePath()` really necessary? This is what I think it describes its intended purpose, https://github.com/ApiGen/ApiGen/blob/4.2/tests/Utils/FileSystemTest.php#L24

### CODE REVIEW

**Comment by luka-zitnik** [comment]
I've just enabled E_DEPRICATED. Will update this patch.

**Comment by ek9** [comment]
Or is this PR to fix issue on windows specifically (as this is marked as a fix for #804 )?

**Comment by ek9** [comment]
I still had problems running tests on windows. Nonetheless, I have incorporates commits from this PR in `appveyor` branch (to enable windows based CI) which also includes further fixes to tests and classes to rely on DIRECTORY_SEPARATOR.

The changes in ths PR are useful and they will be merged as soon as `appveyor` branch is ready. Thank you for your contribution.

File: src/Theme/ThemeConfigPathResolver.php [hunk 1]
```diff
@@ -41,7 +42,7 @@ public function resolve($path)
         ];
 
         foreach ($allowedPaths as $allowedPath) {
-            $absolutePath = $allowedPath . '/' . ltrim($path, DIRECTORY_SEPARATOR);
+            $absolutePath = $allowedPath . '/' . ltrim($path, '/');
```
Comment (luka-zitnik) [inline src/Theme/ThemeConfigPathResolver.php]: This one trims a user-submitted value, which in https://github.com/ApiGen/ApiGen/blob/24842aa702352faf65c1579ea08c1ee118eb0b79/tests/Theme/ThemeConfigPathResolverTest.php#L16 begins with a slash. It depends on its usage, but the function should probably trim both, slash and backslash.

Comment (luka-zitnik) [inline src/Theme/ThemeConfigPathResolver.php]: I think DIRECTORY_SEPARATOR is a matter of preference. Paths in PHP on Windows are regular either way.

That being said, I believe that the constant only makes users spend more time on writing paths, but would commit to that if that's a style rule of a project I'm a part of.

**Comment by luka-zitnik** [comment]
Let me know if there's something that you think needs to be fixed, but you would rather focus on something else instead.

**Comment by ek9** [comment]
This PR was merged as part of #834 PR. This can now be closed, thank you!

**Pull request merged**